compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1360,113 +1263,72 @@ private boolean checkSimilarErrorBindingPatterns(BLangErrorBindingPattern firstE
-                secondSimpleBindingPattern.captureBindingPattern == null)) {
-            return false;
-        }
-
-        if (firstSimpleBindingPattern.wildCardBindingPattern != null
-                && secondSimpleBindingPattern.wildCardBindingPattern == null) {
-            return false;
-        }
-        if (firstSimpleBindingPattern.wildCardBindingPattern == null
-                && secondSimpleBindingPattern.wildCardBindingPattern != null) {
-            return false;
+            return firstSimpleBindingPattern.wildCardBindingPattern != null;
         }
-
-        return true;
+        return firstSimpleBindingPattern == null && secondSimpleBindingPattern == null;
     }
 
     private boolean checkSimilarErrorCauseBindingPattern(BLangErrorCauseBindingPattern firstErrorCauseBindingPattern,
                                                      BLangErrorCauseBindingPattern secondErrorCauseBindingPattern) {
-        if ((firstErrorCauseBindingPattern != null && secondErrorCauseBindingPattern == null)
-                || (firstErrorCauseBindingPattern == null && secondErrorCauseBindingPattern != null)) {
-            return false;
-        }
-        if (firstErrorCauseBindingPattern == null) {
-            return true;
-        }
-        if (!checkSimilarSimpleBindingPattern(firstErrorCauseBindingPattern.simpleBindingPattern,
-                secondErrorCauseBindingPattern.simpleBindingPattern)) {
-            return false;
+        if (firstErrorCauseBindingPattern != null && secondErrorCauseBindingPattern != null) {
+            if (!checkSimilarSimpleBindingPattern(firstErrorCauseBindingPattern.simpleBindingPattern,
+                    secondErrorCauseBindingPattern.simpleBindingPattern)) {
+                return false;
+            }
+            return checkSimilarErrorBindingPatterns(firstErrorCauseBindingPattern.errorBindingPattern,
+                    secondErrorCauseBindingPattern.errorBindingPattern);
         }
-        return checkSimilarErrorBindingPatterns(firstErrorCauseBindingPattern.errorBindingPattern,
-                secondErrorCauseBindingPattern.errorBindingPattern);
+        return firstErrorCauseBindingPattern == null && secondErrorCauseBindingPattern == null;
     }
 
     private boolean checkSimilarErrorFieldBindingPatterns(
                                           BLangErrorFieldBindingPatterns firstErrorFieldBindingPatterns,
                                           BLangErrorFieldBindingPatterns secondErrorFieldBindingPatterns) {
-        if (firstErrorFieldBindingPatterns != null && secondErrorFieldBindingPatterns == null) {
-            return firstErrorFieldBindingPatterns.restBindingPattern != null;
-        }
-        if (firstErrorFieldBindingPatterns == null && secondErrorFieldBindingPatterns != null) {
-            return secondErrorFieldBindingPatterns.restBindingPattern != null;
-        }
         if (firstErrorFieldBindingPatterns == null) {
             return true;
```

Comment:
Is it valid to have this as first logic. Previously even when `firstErrorFieldBindingPatterns` is null we had a chance to return `false` if second rest binding pattern is null isn't it?

Reply:
If our first binding pattern does not have error-field-binding patterns (i.e. `error(MSG, c)`), it matches any error value with matches error-message-binding-pattern and error-cause-binding-pattern.